### PR TITLE
CI adjustments for conda-libmamba-solver as default

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

conda 23.10.0 requires conda-libmamba-solver and has `solver=libmamba` as default, so conda-build's CI is going to be using it now.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Review by mbargull — DISMISSED** [review]
LGTM

**Comment by mbargull** [comment]
@jaimergp, this is still marked as a draft.
Any reason not to merge?

**Comment by jaimergp** [comment]
Just the pre-commit and missing news. I'll commit them now. Thanks for the reminder!

**Review by mbargull — DISMISSED** [review]

**Review by jezdez — APPROVED** [review]

**Comment by jaimergp** [comment]
Errors are related to conda-libmamba-solver itself, probably to be fixed by https://github.com/conda/conda-libmamba-solver/pull/365.

**Comment by jaimergp** [comment]
Let's see how conda-libmamba-solver 23.11.1 does now :)

**Comment by kenodegard** [comment]
@jaimergp no idea why `tests/cli/test_main_build.py::test_build_output_folder` and `tests/cli/test_main_build.py::test_build_source` are failing, attempting to fix it by switching to `tmp_path`, see https://github.com/conda/conda-build/pull/5059/commits/031dac440aa359463cec96f8ad91004dc56f0496

**Comment by jaimergp** [comment]
Me neither @kenodegard 🤷 Thanks for the help though! I'm going to be the solver-related issue `tests/test_api_build.py::test_build_msvc_compiler[14.0] - KeyError: Dist(channel='local', dist_name='conda-build-test-build-msvc-compiler-1.0-0', name='conda-build-test-build-msvc-compiler', fmt='.tar.bz2', version='1.0', build_string='0', build_number=0, base_url=None, platform=None)` in https://github.com/conda/conda-libmamba-solver/issues/386. The other things might be artifacts of the solver now being faster, which means things need to be recreated more quickly than before? Idk. If feels like a race condition.

**Comment by mbargull** [comment]
> @jaimergp no idea why `tests/cli/test_main_build.py::test_build_output_folder` and `tests/cli/test_main_build.py::test_build_source` are failing (see https://github.com/conda/conda-build/actions/runs/6897842353/job/18770690271?pr=5059#step:8:3164), attempting to fix it by switching to `tmp_path`, see [031dac4](https://github.com/conda/conda-build/commit/031dac440aa359463cec96f8ad91004dc56f0496)

This is most likely fixed by gh-5068 .
What we have here is that the `croot` is being modified and/or removed by concurrently running tests which is why `conda-index` fails.

**Comment by mbargull** [comment]
It would be good if we could merge this PR and gh-5068 to remove some of the failing test clutter.
Further fixes should be done/investigated in subsequent PRs.
@jaimergp, mark as "Ready for review"?

**Comment by mbargull** [comment]
> > @jaimergp no idea why `tests/cli/test_main_build.py::test_build_output_folder` and `tests/cli/test_main_build.py::test_build_source` are failing (see https://github.com/conda/conda-build/actions/runs/6897842353/job/18770690271?pr=5059#step:8:3164), attempting to fix it by switching to `tmp_path`, see [031dac4](https://github.com/conda/conda-build/commit/031dac440aa359463cec96f8ad91004dc56f0496)
> 
> This is most likely fixed by [gh-5068](https://github.com/conda/conda-build/pull/5068) . What we have here is that the `croot` is being modified and/or removed by concurrently running tests which is why `conda-index` fails.

Ah, taking another look at those 2 tests, I see that they use `--croot` explicitly; so those might not necessarily be affected by gh-5068. (We should merge that in nonetheless.)

**Comment by jaimergp** [comment]
Agreed @mbargull. Thanks!

**Review by jezdez — APPROVED** [review]

**Comment by jezdez** [comment]
@mbargull Just approved, but I wanted to clarify that it's a-ok to merge given the new build tools team effort. 🚀

**Comment by mbargull** [comment]
> @mbargull Just approved, but I wanted to clarify that it's a-ok to merge given the new build tools team effort. 🚀

:heart: Thanks.
(I've been out for so long, I'm only slowly getting the lay of the land of the current state of who/what/when in Conda land ;).)

File: tests/test_api_build.py [hunk 1]
```diff
@@ -123,6 +124,9 @@ def test_recipe_builds(
     #   ``source_setup_py_data_subdir`` reproduces the problem.
     if recipe.name == "source_setup_py_data_subdir":
         pytest.xfail("Issue related to #3754 on conda-build.")
+    elif recipe.name == "unicode_all_over" and context.solver == "libmamba":
+        pytest.xfail("Unicode package names not supported in libmamba.")
```
Comment (dholth) [inline tests/test_api_build.py:128]: Yikes

**Review by jezdez — APPROVED** [review]

**Review by kenodegard — APPROVED** [review]

File: tests/test_api_build.py (hunk 1 above)
Comment (jezdez) [reply tests/test_api_build.py:128]: libmamba doesn't support unicode package names IIUC, no need to test for it.

**Pull request merged**